internal/bminventory/inventory.go

```diff
@@ -5359,8 +5359,17 @@ func (b *bareMetalInventory) V2RegisterHost(ctx context.Context, params installe
 
 	b.customizeHost(c, host)
 
-	eventgen.SendHostRegistrationSucceededEvent(ctx, b.eventsHandler, *params.NewHostParams.HostID,
-		params.InfraEnvID, host.ClusterID, hostutil.GetHostnameForMsg(host))
+	// Only send SendHostRegistrationSucceededEvent if cluster in a pre-installation state.
+	postInstallStages := []string{
```

Comment:
did you consider to move this event into https://github.com/openshift/assisted-service/blob/f218f6c0bcb729f5bca66608f357cd7401b2363a/internal/host/transition.go#L85 
then you will be able to avoid looking at specific states

Reply:
I considered moving it but I think a lot of our tests are structured in a way that would require a larger rewrite in order to prove the event emission. I felt it was more concise to leave things as they are.